test/dlc_tests/conftest.py

```diff
@@ -291,15 +312,207 @@ def ec2_instance_role_name(request):
 
 
 @pytest.fixture(scope="function")
-def ec2_instance_ami(request):
-    return request.param if hasattr(request, "param") else UBUNTU_18_BASE_DLAMI_US_WEST_2
+def ec2_instance_ami(request, region):
+    return (
+        request.param
+        if hasattr(request, "param")
+        else UBUNTU_18_BASE_DLAMI_US_EAST_1
+        if region == "us-east-1"
+        else UBUNTU_18_BASE_DLAMI_US_WEST_2
+    )
 
 
 @pytest.fixture(scope="function")
 def ei_accelerator_type(request):
     return request.param if hasattr(request, "param") else None
 
 
+@pytest.mark.timeout(300)
+@pytest.fixture(scope="function")
+def efa_ec2_instances(
+    request,
+    ec2_client,
+    ec2_resource,
+    ec2_instance_type,
+    ec2_instance_role_name,
+    ec2_key_name,
+    ec2_instance_ami,
+    region,
+    availability_zone_options,
+):
+    _validate_p4de_usage(request, ec2_instance_type)
+    ec2_key_name = f"{ec2_key_name}-{str(uuid.uuid4())}"
+    print(f"Creating instance: CI-CD {ec2_key_name}")
+    key_filename = test_utils.generate_ssh_keypair(ec2_client, ec2_key_name)
+
+    def delete_ssh_keypair():
+        if test_utils.is_pr_context():
+            test_utils.destroy_ssh_keypair(ec2_client, key_filename)
+        else:
+            with open(KEYS_TO_DESTROY_FILE, "a") as destroy_keys:
+                destroy_keys.write(f"{key_filename}\n")
+
+    request.addfinalizer(delete_ssh_keypair)
+
+    instance_name_prefix = f"CI-CD {ec2_key_name}"
```

Comment:
nit: we can add `EFA` to the name be be explicit

Reply:
The test name would have EFA in it already. Also, the instances launched would be renamed with `master` and `worker` suffixes. Is there any other reason we might need `efa` in the name?

Reply:
if efa is already in the test name, that would work